pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,559 @@
+
+    waveform_points_mode = Instrument.control(
+        ":waveform:points:mode?", ":waveform:points:mode %s",
+        """ A string parameter that sets the data record to be transferred with the waveform_data
+         method. Can be "normal", "maximum", or "raw".""",
+        validator=strict_discrete_set,
+        values={"normal": "NORM", "maximum": "MAX", "raw": "RAW"},
+        map_values=True
+    )
+    waveform_points = Instrument.control(
+        ":waveform:points?", ":waveform:points %d",
+        """ An integer parameter that sets the number of waveform points to be transferred with
+        the waveform_data method. Can be any of the following values: 
+        100, 250, 500, 1000, 2 000, 5 000, 10 000, 20 000, 50 000, 62 500.
+        
+        Note that the oscilloscope may provide less than the specified nb of points. """,
+        validator=strict_discrete_set,
+        values=[100, 250, 500, 1000, 2000, 5000, 10000, 20000, 50000, 62500]
+    )
+    waveform_source = Instrument.control(
+        ":waveform:source?", ":waveform:source %s",
+        """ A string parameter that selects the analog channel, function, or reference waveform 
+        to be used as the source for the waveform methods. Can be "channel1", "channel2", "function", 
+        "fft", "wmemory1", "wmemory2", or "ext".""",
+        validator=strict_discrete_set,
+        values={"channel1": "CHAN1", "channel2": "CHAN2", "function": "FUNC", "fft": "FFT",
+                "wmemory1": "WMEM1", "wmemory2": "WMEM2", "ext": "EXT"},
+        map_values=True
+    )
+    waveform_format = Instrument.control(
+        ":waveform:format?", ":waveform:format %s",
+        """ A string parameter that controls how the data is formatted when sent from the 
+        oscilloscope. Can be "ascii".""",
+        validator=strict_discrete_set,
+        values={"ascii": "ASC"},
+        map_values=True
+    )
+    @property
+    def waveform_preamble(self):
+        """ Get preamble information for the selected waveform source as a dict with the following keys:
+            - "format": byte, word, or ascii (str)
+            - "type": normal, peak detect, or average (str)
+            - "points": nb of data points transferred (int)
+            - "count": always 1 (int)
+            - "xincrement": time difference between data points (float)
+            - "xorigin": first data point in memory (float)
+            - "xreference": data point associated with xorigin (int)
+            - "yincrement": voltage difference between data points (float)
+            - "yorigin": voltage at center of screen (float)
+            - "yreference": data point associated with yorigin (int)"""
+        return self._waveform_preamble()
+
+    @property
+    def waveform_data(self):
+        """ Get the binary block of sampled data points transmitted using the IEEE 488.2 arbitrary
+        block data format."""
+        # Other waveform formats raise UnicodeDecodeError
```

Comment:
Hm, you can try to use [Instrument.binary_values](https://github.com/ralph-group/pymeasure/blob/c0e3ff0b4ac2bce58639e5410da4af4c6147dddb/pymeasure/instruments/instrument.py#L113) to get the reply correctly (it also can strip headers and convert datatype)? (I have to find out why this is not in the documentation)

Reply:
Was that a suggestion you considered?

Reply:
I've decided to keep the format of data requested by this method to "ascii", which means that we do not received binary values, so no need to use binary_values.